### PR TITLE
PR: Add support to build pull request previews with Netlify

### LINKED ISSUE #150: Create previews of the Pull Requests
## Issue Description

Use something to build PR previews i.e Netlify

### PR DESCRIPTION
**Pull request by ccordoba12**
Fixes #150.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, this one is ready. Please review it.

**Comment by ccordoba12** [comment]
Thanks @CAM-Gerlach for the feedback!

**Comment by CAM-Gerlach** [comment]
That would be very helpful @dalthviz ; I assume that is set up mostly externally on the Netlify site?

**Comment by dalthviz** [comment]
Yep I think so @CAM-Gerlach

**Comment by ccordoba12** [comment]
Yes, I totally agree. I'll add a Markdown file with the steps I followed to configure Netlify.

**Comment by ccordoba12** [comment]
Hey guys, I just added a Markdown file explaining how to configure Netlify. About the two initial steps, please see this image from the Netlify website:

Selección_006

About the second note, please see this one:

Selección_007

Since it's not possible to add more members to a Netlify team without paying 15$ per month, I was thinking to use the Spyder Bot account there, so everyone can configure it in the future if needed. What do you think?

**Comment by ccordoba12** [comment]
Ok, please don't merge until I change the account for Netlify, ok? I think that's the best option going forward.

**Comment by ccordoba12** [comment]
@CAM-Gerlach, could you review the text I added in `Configure Netlify.md`?

**Comment by CAM-Gerlach** [comment]
@ccordoba12 Sure, standby.

**Comment by ccordoba12** [comment]
> Ok, please don't merge until I change the account for Netlify, ok? I think that's the best option going forward.

This is done. So things now depend on @CAM-Gerlach's review.

**Comment by CAM-Gerlach** [comment]
I'll review this as soon as I'm finally finished with #167 , sorry for the delay

**Comment by ccordoba12** [comment]
No problem @CAM-Gerlach

**Comment by ccordoba12** [comment]
@CAM-Gerlach, I moved what I wrote about configuring Netlify to our Contributing guide, as you suggested.

**Comment by CAM-Gerlach** [comment]
Okay, I'll do it first thing in the morning. Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -108,6 +108,27 @@ resulting ``_build`` directory it produces; ``index.html`` is the main page.
...
+
+* Pass the root directory of the generated docs. In our case that is
+
+      doc/_build/html/
+
+*NOTES*:
+
+* The `runtime.txt` file in the root of the repo is needed by Netlify to declare the Python version required to build the docs.
+* By default Netlify adds a lot of checks to pull requests, besides the one that actually allows to preview their changes live.
+To remove those extra checks, you need to go to the `Build and deploy` configuration entry, then to the `Outgoing notifications` section, and remove all commit notifications that don't start by `Add Deploy Preview`.
```
Comment (CAM-Gerlach) [inline CONTRIBUTING.md:128]:
```suggestion
  To remove those extra checks, you need to go to the `Build and deploy` configuration entry, then to the `Outgoing notifications` section, and remove all commit notifications that don't start with `Add Deploy Preview`.
```

Fix grammar and make indent match its corresponding bullet.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -108,6 +108,27 @@ resulting ``_build`` directory it produces; ``index.html`` is the main page.
...
+      ci/install.sh && make docs
+
+* Pass the root directory of the generated docs. In our case that is
+
+      doc/_build/html/
+
+*NOTES*:
+
+* The `runtime.txt` file in the root of the repo is needed by Netlify to declare the Python version required to build the docs.
+* By default Netlify adds a lot of checks to pull requests, besides the one that actually allows to preview their changes live.
```
Comment (CAM-Gerlach) [inline CONTRIBUTING.md:127]:
```suggestion
* By default, Netlify adds a lot of checks to pull requests, besides the one that actually builds the live preview.
```

Fixed some grammar issues and clarified the phrasing

Comment (CAM-Gerlach) [reply CONTRIBUTING.md:127]: Are you sure none of these checks are useful for us?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -108,6 +108,27 @@ resulting ``_build`` directory it produces; ``index.html`` is the main page.
...
+
+The steps to configure Netlify to preview pull requests are the following:
+
+* Pass a command to build the docs. In our case that command is
+
+      ci/install.sh && make docs
+
+* Pass the root directory of the generated docs. In our case that is
+
+      doc/_build/html/
```
Comment (CAM-Gerlach) [inline CONTRIBUTING.md:122]:
~~~suggestion
1. Pass a command to build the docs:

```bash
ci/install.sh && make docs
```

2. Pass the root directory of the generated docs:

```bash
doc/_build/html/
```
~~~

I'm guessing this should be an ordered list. Also, used standard GFM literal blocks for consistency with the rest of the guide and to add bash syntax highlighting, and simplified the sentences a little.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -108,6 +108,27 @@ resulting ``_build`` directory it produces; ``index.html`` is the main page.
 **NOTE:** You can try to use the `make` commands on Windows if you have `make` installed and something like `Cygwin`, `MSYS2` or `MYSYS`, or by using Windows Subsystem for Linux (`WSL`).
 
 
+
+## Configuring Netlify
+
+The steps to configure Netlify to preview pull requests are the following:
```
Comment (CAM-Gerlach) [inline CONTRIBUTING.md:114]: I was a little confused...where do these steps go, exactly? Just copying and pasting the exact literals into some field on the Netlify site, or what?

Comment (ccordoba12) [reply CONTRIBUTING.md:114]: Yes, they need to be declared in the Netlify site.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ccordoba12) [reply CONTRIBUTING.md:127]: We can check later.

**Pull request merged**